corehq/apps/app_manager/views/modules.py

```diff
@@ -1588,8 +1589,11 @@ def new_module(request, domain, app_id):
                 followup = app.new_form(module_id, _("Followup Form"), lang, attachment=attachment)
                 followup.requires = "case"
                 followup.actions.update_case = UpdateCaseAction(condition=FormActionCondition(type='always'))
-
-            _init_module_case_type(module)
+            case_type = request.POST.get('case_type', None)
+            if case_type:
+                module.case_type = case_type
+            else:
+                _init_module_case_type(module)
```

Comment:
Could we incorporate setting the case type into `_init_module_case_type`? The name `_init_module_case_type` suggests that this function would also handle this behavior.

Reply:
919ca87e83202cfa47217e44a5986623d1b5b163